instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/patches/producer.rb

```diff
@@ -17,9 +17,18 @@ def produce(value, topic:, key: nil, headers: {}, partition: nil, partition_key:
               'messaging.destination_kind' => 'topic'
             }
 
-            tracer.in_span("#{topic} send", attributes: attributes, kind: :producer) do
-              OpenTelemetry.propagation.inject(headers)
-              super
+            # If context is unset, try to inject headers injected by async producer
+            ctx = if OpenTelemetry::Trace.current_span == OpenTelemetry::Trace::Span::INVALID
```

Comment:
I think so: 00cd9e8.

Now, logic is: if there's already a span recording when we produce, use that as context; otherwise try to extract context from headers.

There's maybe an edge case where the active span when #producer runs has been sampled out, but produce is being called with headers: with a trace context indicating a sampled in parent. Should we (info-level) log a warning in that circumstance?

Reply:
> Should we (info-level) log a warning in that circumstance?

What would the info log message say?

Reply:
Sorry, I totally missed this @robbkidd! I've changed the code a bit since I made this comment so I don't think a log is necessary...